### ISSUE TITLE
Configurable hotadd feature in vSphere 6.7

### ISSUE THREAD

**Issue by samzehnder** [apache/cloudstack#4472]
##### ISSUE TYPE
 * Improvement Request
 * Enhancement Request
 * Feature Idea

##### COMPONENT NAME

~~~
Configurable VMware Config Spec for hotadd actions in virtual machines in VMware 6.7+ environments.
~~~

##### CLOUDSTACK VERSION

~~6.14~~ , I suppose 4.14 is meant here, but new features are not implemented on released versions, so this will be 4.15, probably.

##### CONFIGURATION

vSphere 6.7

##### OS / ENVIRONMENT

N/A

##### SUMMARY

Some virtual machines should not be configured with hotadd support. Before vSphere 6.7 it is possible to overwrite the config via instance details, which are then translated to vmware extraOptionConfig (i.e. vcpu.hotadd=false).

Since vSphere 6.7 this is no longer supported, as the hotadd feature is not an extraOption anymore, but a vmSpecConfig, and this get's set by CloudStack based on OS-type (s. [VmwareResource.java](https://github.com/apache/cloudstack/blob/7f408ec967327a152d903dcc55260ccff9bad0fd/plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java#L1961))

General idea is, to use the instance details to control the settings, i.e. if vcpu.hotadd=false is set in instance details, then set hotadd feature for vCPU to false as well in vmConfigSpec.

##### STEPS TO REPRODUCE

~~~
Instance detail is added with key=>value: vcpu.hotadd=>false
~~~

##### EXPECTED RESULTS

~~~
Above details get's translated to VM Spec Config: vmConfigSpec.setCpuHotAddEnabled(False)
~~~

##### ACTUAL RESULTS

~~~
Setting hotadd feature is only based on guest os descriptor.
~~~

**Comment by DaanHoogland** [comment]
@PaulAngus @andrijapanicsb care to discuss?

**Comment by nvazquez** [comment]
The proposed idea looks feasible to me for non-deploy-as-is VMs, @rhtyd @DaanHoogland @andrijapanicsb what do you think about depoy-as-is VMs?

**Comment by DaanHoogland** [comment]
wouldn't do @nvazquez , we will have to alter a (no longer) deploy-as-is template. I have slight moral objections to that. For other templates it seems like a maintenance issue.

**Comment by harikrishna-patnala** [comment]
@samzehnder, we already have dynamicallyScalable property of the VM which can be used to on and off hotplug on VM. There is a PR https://github.com/apache/cloudstack/pull/4630 which addresses this specifically for VMware. Or am I missing anything here !

**Comment by nvazquez** [comment]
Closing this as has been fixed by #4630
